src/main/java/net/mcreator/ui/dialogs/workspace/GeneratorSelector.java

```diff
@@ -60,6 +73,9 @@ public static GeneratorConfiguration getGeneratorSelector(Window parent, @Nullab
 
 			if (currentFlavor == null || currentFlavor.equals(generatorConfiguration.getGeneratorFlavor())) {
 				generator.addItem(generatorConfiguration);
+			} else if (((currentFlavor == GeneratorFlavor.FORGE && generatorConfiguration.getGeneratorFlavor() == GeneratorFlavor.FABRIC) ||
+					(currentFlavor == GeneratorFlavor.FABRIC && generatorConfiguration.getGeneratorFlavor() == GeneratorFlavor.FORGE)) && !newWorkspace) {

```

Comment:
You can remove some odd brackets here:
<kbd>((forge && fabric) || (fabric && forge)) && !new</kbd>
`->`
<kbd>(forge && fabric || fabric && forge) && !new</kbd>